pkg/agent/agent_windows.go

```diff
@@ -369,21 +370,16 @@ func (i *Initializer) getTunnelPortLocalIP() net.IP {
 // The routes will be restored on OVS bridge interface after the IP configuration
 // is moved to the OVS bridge.
 func (i *Initializer) saveHostRoutes() error {
-	routes, err := util.GetNetRoutesAll()
+	family := antreasyscall.AF_INET
+	filter := &util.Route{
+		LinkIndex:      i.nodeConfig.UplinkNetConfig.Index,
+		GatewayAddress: net.ParseIP(i.nodeConfig.UplinkNetConfig.Gateway),
+	}
+	routes, err := util.RouteListFiltered(family, filter, util.RT_FILTER_IF|util.RT_FILTER_GW)
 	if err != nil {
 		return err
 	}
 	for _, route := range routes {
-		if route.LinkIndex != i.nodeConfig.UplinkNetConfig.Index {
-			continue
-		}
-		if route.GatewayAddress.String() != i.nodeConfig.UplinkNetConfig.Gateway {
-			continue
-		}
-		// Skip IPv6 routes before we support IPv6 stack.
```

Comment:
How is IPv6 stack handled after the change?

Reply:
Use the following code to get IPv6 routes:

```go
routes, err := util.RouteListFiltered(antreasyscall.AF_INET6, filter, util.RT_FILTER_IF|util.RT_FILTER_GW)
```

Reply:
IIRC, we don't support IPv6 on Windows right now. If we support IPv6 in the future, we could change the code like this to get dual stack routes:

```go
family := antreasyscall.AF_UNSPEC
	filter := &util.Route{
		LinkIndex:      i.nodeConfig.UplinkNetConfig.Index,
		GatewayAddress: net.ParseIP(i.nodeConfig.UplinkNetConfig.Gateway),
	}
	routes, err := util.RouteListFiltered(family, filter, util.RT_FILTER_IF|util.RT_FILTER_GW)
```